client/hostvolumemanager/host_volumes.go

```diff
@@ -63,14 +105,22 @@ func (hvm *HostVolumeManager) Create(ctx context.Context,
 		return nil, err
 	}
 
+	volState := &HostVolumeState{
+		ID:        req.ID,
+		CreateReq: req,
+	}
+	if err := hvm.stateMgr.PutDynamicHostVolume(volState); err != nil {
+		hvm.log.Error("failed to save volume in state", "volume_id", req.ID, "error", err)
+		// db TODO: bail or nah?
```

Comment:
If we create the volume on-disk but not in client state / server state, we'll end up with stray volumes we have no way of cleaning up. I'd try to implement a transaction-like semantic here -- if we can't fully complete the task, we should destroy the volume when we're done.